v-next/hardhat/src/internal/core/global-options.ts

```diff
@@ -80,7 +80,7 @@ export function buildGlobalOptionDefinition<T extends ArgumentType>({
   type?: T;
   defaultValue: ArgumentTypeToValueType<T>;
 }): OptionDefinition {
-  const argumentType = type ?? ArgumentType.STRING;
+  const argumentType = type ?? ArgumentType.STRING_WITHOUT_DEFAULT;
```

Comment:
This would go away if we remove the optionality on type for global options.
If we don't think it should be optional should we keep it as `ArgumentType.string`?

Reply:
We are keeping type optionality and setting the type of `T` to `STRING` explicitly, so this should be reverted so that string is the runtime default as well